lib/request.js

```diff
@@ -292,8 +291,7 @@ module.exports = {
    */
 
   get idempotent() {
-    const methods = ['GET', 'HEAD', 'PUT', 'DELETE', 'OPTIONS', 'TRACE'];
-    return !!~methods.indexOf(this.method);
+    return ['GET', 'HEAD', 'PUT', 'DELETE', 'OPTIONS', 'TRACE'].includes(this.method);
```

Comment:
@PlasmaPower I wrote that it will improve performance, but drop readability. Giving in account our conversation regarding importance of readability earlier, I'm afraid to do such changes in a small method of external API that doesn't affect Koa performance in any way. But I'm quite OK to move static arrays into constants if needed.